src/EndpointV2/Ruleset/RulesetStandardLibrary.php

```diff
@@ -0,0 +1,431 @@
+    }
+
+    /**
+     * Evaluates two booleans for equality.
+     *
+     * @return boolean
+     */
+    public function booleanEquals($boolean1, $boolean2)
+    {
+        return
+            filter_var($boolean1, FILTER_VALIDATE_BOOLEAN)
+            === filter_var($boolean2, FILTER_VALIDATE_BOOLEAN);
+    }
+
+    /**
+     * Percent-encodes an input string.
+     *
+     * @return mixed
+     */
+    public function uriEncode($input)
+    {
+        if (is_null($input)) {
+            return null;
+        }
+        return str_replace('%7E', '~', rawurlencode($input));
+    }
+
+    /**
+     * Parses URL string into components.
+     *
+     * @return mixed
+     */
+    public function parseUrl($url)
+    {
+        if (is_null($url)) {
+            return null;
+        }
+
+        $parsed = parse_url($url);
+
+        if ($parsed === false || !empty($parsed['query'])) {
+            return null;
+        } else if (isset($parsed['scheme'])) {
+            if ($parsed['scheme'] !== 'http'
+                && $parsed['scheme'] !== 'https'
+            ) {
+                return null;
+            }
+        }
+
+        $urlInfo = [];
+        $urlInfo['scheme'] = isset($parsed['scheme']) ? $parsed['scheme'] : '';
+        $urlInfo['authority'] = isset($parsed['host']) ? $parsed['host'] : '';
+        if (isset($parsed['port'])) {
+            $urlInfo['authority'] = $urlInfo['authority'] . ":" . $parsed['port'];
+        }
+        $urlInfo['path'] = isset($parsed['path']) ? $parsed['path'] : '';
+        $urlInfo['normalizedPath'] = !empty($parsed['path'])
+            ? rtrim($urlInfo['path'] ?: '', '/' .  "/") . '/'
+            : '/';
+        $urlInfo['isIp'] = !isset($parsed['host']) ?
+            'false' : $this->isValidIp($parsed['host']);
+
+        return $urlInfo;
+    }
+
+    /**
+     * Evaluates whether a value is a valid host label per
+     * RFC 1123. If allow_subdomains is true, split on `.` and validate
+     * each subdomain separately.
+     *
+     * @return boolean
+     */
+    public function isValidHostLabel($hostLabel, $allowSubDomains)
+    {
+        if (!isset($hostLabel)
+            || (!$allowSubDomains && strpos($hostLabel, '.') != false)
+        ) {
+            return false;
+        }
+
+        if ($allowSubDomains) {
+            foreach (explode('.', $hostLabel) as $subdomain) {
+                if (!$this->validateHostLabel($subdomain)) {
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return $this->validateHostLabel($hostLabel);
+        }
+    }
+
+    /**
+     * Parse and validate string for ARN components.
+     *
+     * @return array|null
+     */
+    public function parseArn($arnString)
+    {
+        if (is_null($arnString)
+            || substr( $arnString, 0, 3 ) !== "arn"
+        ) {
+            return null;
+        }
+
+            $arn = [];
+        $parts = explode(':', $arnString, 6);
```

Comment:
Is this directly from the SEP?  I feel like this would break in something like an IAM arn where there's two colons in a row: `arn:aws:iam::123456789012:user`.

Reply:
This isn't explicitly called out in the spec, but from what I'm seeing IAM will accept the standard format as well. The current formats we allow in the arninterface are: 

`arn:partition:service:region:account-id:resource-id`
`arn:partition:service:region:account-id:resource-type/resource-id`
`arn:partition:service:region:account-id:resource-type:resource-id`

I had to hand-roll this rather than use our built-in arn parser because object lambda arns were not making through the initial validation.